client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.client.opentracing;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+import org.hawkular.apm.api.model.Property;
+
+import io.opentracing.APMSpan;
+
+/**
+ * This node processor implementation provides the default mapping behaviour from
+ * the span's tags.
+ *
+ * @author gbrown
+ */
+public class DefaultNodeProcessor implements NodeProcessor {
+
+    /* (non-Javadoc)
+     * @see org.hawkular.apm.client.opentracing.NodeProcessor#process(io.opentracing.APMSpan, org.hawkular.apm.client.opentracing.NodeBuilder)
+     */
+    @Override
+    public void process(APMSpan span, NodeBuilder nodeBuilder) {
+        for (Map.Entry<String, Object> entry : span.getTags().entrySet()) {
+            nodeBuilder.addProperty(new Property(entry.getKey(), entry.getValue().toString()));
+
+            if (entry.getKey().endsWith(".url") || entry.getKey().endsWith(".uri")) {
+                try {
+                    URL url = new URL(entry.getValue().toString());
+                    nodeBuilder.setUri(url.getPath());
```

Comment:
Yes, that is because when the model was first created, it was a URI (including host/port) - according to the definition it is a "string of characters to locate a resource". However including the host/port made it difficult to aggregate across clustered services, and also the concept needed to be more generic to cope with other types of endpoint or even components.

So we may want to reconsider the name, but generally speaking it is a resource id.